lib/project.ts

```diff
@@ -368,22 +373,13 @@ export class Project implements Project.IProject {
 	}
 
 	public getConfigurationsSpecifiedByUser(): string[] {
-		let configurations: string[] = [];
-		if (this.$options.debug) {
-			configurations.push(this.$projectConstants.DEBUG_CONFIGURATION_NAME);
-		}
-
-		if (this.$options.release) {
-			configurations.push(this.$projectConstants.RELEASE_CONFIGURATION_NAME);
+		return this.$options.config || [];
 		}
 
-		return configurations;
-	}
-
 	public updateProjectPropertyAndSave(mode: string, propertyName: string, propertyValues: string[]): IFuture<void> {
 		return (() => {
 			this.ensureProject();
-			let projectConfigurations = _.keys(this.configurationSpecificData);
+			let projectConfigurations = this.getConfigurationsSpecifiedByUser();
```

Comment:
If the user enters invalid configuration we should display warning which says that the invalid configuration does not exist and the changes for it will not be applied or throw an error with message for invalid configuration. Now if an invalid configuration is passed the command will continue and print successful message but there will be no result from the command.